### ISSUE TITLE
Catapult server fails to create a block with maximize-fee strategy

### ISSUE THREAD

**Issue by Wayonb** [nemtech/catapult-server#84]
The below code will trigger the issue.

transactionSelectionStrategy = maximize-fee

````
        final Account B = Account.generateNewAccount(testContext.getNetworkType());
        final Account C = Account.generateNewAccount(testContext.getNetworkType());
        transferHelper.submitTransferAndWait(sender, B.getAddress(),
                Arrays.asList(testContext.getNetworkCurrency().createRelative(BigInteger.valueOf(100))), PlainMessage.Empty);

        transferHelper.withMaxFee(BigInteger.valueOf(200000)).createTransferAndAnnounce(sender, B.getAddress(),
                Arrays.asList(testContext.getNetworkCurrency().createRelative(BigInteger.valueOf(100))), PlainMessage.Empty);
        new TransferHelper(testContext).withMaxFee(BigInteger.valueOf(200000)).createTransferAndAnnounce(B, C.getAddress(),
                Arrays.asList(testContext.getNetworkCurrency().createRelative(BigInteger.valueOf(20))), PlainMessage.Empty);
        new TransferHelper(testContext).withMaxFee(BigInteger.valueOf(600000)).createTransferAndAnnounce(C, B.getAddress(),
                Arrays.asList(testContext.getNetworkCurrency().createRelative(BigInteger.valueOf(5))), PlainMessage.Empty);
````

````
peer-node-0_1        | 2020-08-20 04:20:20.019902 0x00007fbe56ffd700: <debug> (utils::StackLogger.h@36) pushing scope 'generating candidate block' 
peer-node-0_1        | 2020-08-20 04:20:20.020844 0x00007fbe56ffd700: <debug> (utils::StackLogger.h@42) popping scope 'generating candidate block' (0ms) 
peer-node-0_1        | 2020-08-20 04:20:20.020883 0x00007fbe56ffd700: <info> (src::ScheduledHarvesterTask.cpp@35) successfully harvested block at 303 with signer 365764D3C089A9AEB8E9C1C68E836A56D0406C2A16E766B59479203F5CABF9F1 
peer-node-0_1        | 2020-08-20 04:20:20.020919 0x00007fbe56ffd700: <debug> (disruptor::Disruptor.cpp@42) disruptor queuing element 788 (1 blocks (heights 303 - 303) [00000000] from Local) 
peer-node-0_1        | 2020-08-20 04:20:20.073921 0x00007fbdfc7f8700: <warning> (consumers::BlockChainProcessor.cpp@220) block state hash (5F45A715CAD324024B4F6FE44FDDD82B487217CA0FDF88AD836D35E5EEE75018) does not match cache state hash (362CE6F9123D8964F09516BCC984C617BA5142039AC7B5D9E107365EF27F6130) at height 303 
peer-node-0_1        | 2020-08-20 04:20:20.073960 0x00007fbdfc7f8700: <info> (consumers::BlockChainProcessor.cpp@226) cache state hash (9 components) at height 302
peer-node-0_1        | 9443C34887626CC4BE5439B4A4435DE380EE644EECE1C78FE02DF0EF7438E3D7
peer-node-0_1        |  + 2571DF34FB742FAEB5EC7FD0F3D15B4E1037670282A8C57768D2FB6E3CC518C9
peer-node-0_1        |  + 8C16AFF06896D4EA6290AEC7285F9350F27BDEC71DFD3328475709CDF30C6795
peer-node-0_1        |  + 31F4A7C7F94153CE4E5634C08AE4DEDEB0C1B30BDE43A37E5255664F17D96DCC
peer-node-0_1        |  + 0000000000000000000000000000000000000000000000000000000000000000
peer-node-0_1        |  + 0000000000000000000000000000000000000000000000000000000000000000
peer-node-0_1        |  + 0000000000000000000000000000000000000000000000000000000000000000
peer-node-0_1        |  + 0000000000000000000000000000000000000000000000000000000000000000
peer-node-0_1        |  + 0000000000000000000000000000000000000000000000000000000000000000
peer-node-0_1        |  + 0000000000000000000000000000000000000000000000000000000000000000 
peer-node-0_1        | 2020-08-20 04:20:20.073967 0x00007fbdfc7f8700: <info> (consumers::BlockChainProcessor.cpp@226) cache state hash (9 components) at height 303
peer-node-0_1        | 362CE6F9123D8964F09516BCC984C617BA5142039AC7B5D9E107365EF27F6130
peer-node-0_1        |  + 8502114DE32BE1D36FD1E9924F265DB0459022A72AA9AA953CB20A8654CB928E
peer-node-0_1        |  + 8C16AFF06896D4EA6290AEC7285F9350F27BDEC71DFD3328475709CDF30C6795
peer-node-0_1        |  + 31F4A7C7F94153CE4E5634C08AE4DEDEB0C1B30BDE43A37E5255664F17D96DCC
peer-node-0_1        |  + 0000000000000000000000000000000000000000000000000000000000000000
peer-node-0_1        |  + 0000000000000000000000000000000000000000000000000000000000000000
peer-node-0_1        |  + 0000000000000000000000000000000000000000000000000000000000000000
peer-node-0_1        |  + 0000000000000000000000000000000000000000000000000000000000000000
peer-node-0_1        |  + 0000000000000000000000000000000000000000000000000000000000000000
peer-node-0_1        |  + 0000000000000000000000000000000000000000000000000000000000000000 
peer-node-0_1        | 2020-08-20 04:20:20.073975 0x00007fbdfc7f8700: <warning> (consumers::BlockChainSyncConsumer.cpp@246) processing of peer chain failed with Failure_Chain_Block_Inconsistent_State_Hash 
peer-node-0_1        | 2020-08-20 04:20:20.087848 0x00007fbdfb7f6700: <info> (disruptor::ConsumerDispatcher.cpp@43) completing processing of element 788 (1 blocks (heights 303 - 303) [705652F2] from Local), last consumer is 0 elements behind 
````

**Comment by Wayonb** [comment]
These are the transactions in the ut cache

{
  "data": [
    {
      "meta": {
        "height": "0",
        "hash": "63608D405ED325BD3074C3F0EF1D4E84309DE69DDD2D4B45812887BBBB12026B",
        "merkleComponentHash": "63608D405ED325BD3074C3F0EF1D4E84309DE69DDD2D4B45812887BBBB12026B",
        "index": 0
      },
      "transaction": {
        "size": 177,
        "signature": "DD49D9284CEA45E78D40D13F6EB0494CA259A2BCA96ADD71D31282824BD0E0664E16DF427933AD2B7A1DE305DE2907C14AC581D5F0996DDD7F92E70206F3DB0A",
        "signerPublicKey": "7E6E7EC7D9BE290C371DB107AC6D1AA4143313CAA68F73BB0F73AF1E14849484",
        "version": 1,
        "network": 152,
        "type": 16724,
        "maxFee": "200000",
        "deadline": "24473627692",
        "recipientAddress": "9825CEC20D518AE310E38ECC4CF9297F2624CEC9937A1DE8",
        "message": {
          "type": 0,
          "payload": ""
        },
        "mosaics": [
          {
            "id": "243661FECB8BE288",
            "amount": "20000000"
          }
        ]
      },
      "id": "5F3DF87C75AE7B73B969FF67"
    },
    {
      "meta": {
        "height": "0",
        "hash": "41CFB2EAA20D867F96578D41EA884DF4575805B0D2BCA8163756D57A857FF646",
        "merkleComponentHash": "41CFB2EAA20D867F96578D41EA884DF4575805B0D2BCA8163756D57A857FF646",
        "index": 0
      },
      "transaction": {
        "size": 177,
        "signature": "052E92601B68CBF22C3F9E356FF0A47564D1ACDAA48817D66FE17F88C5D29AC900B0CE9BC2732E38885C4FDAFAAD1B8A1906C71E3631B839A18EF455BC5FB20F",
        "signerPublicKey": "EF7AD177FAB80BA2D481CF16D59974A40C800F8B28960B6A14C3188930AAE3EB",
        "version": 1,
        "network": 152,
        "type": 16724,
        "maxFee": "200000",
        "deadline": "24473627685",
        "recipientAddress": "9815D90D4F30FC2FEAD12A67D9DAB88BB85A8AEAA5D2E589",
        "message": {
          "type": 0,
          "payload": ""
        },
        "mosaics": [
          {
            "id": "243661FECB8BE288",
            "amount": "100000000"
          }
        ]
      },
      "id": "5F3DF87C75AE7B73B969FF68"
    },
    {
      "meta": {
        "height": "0",
        "hash": "1C01452EB85B4BE3F1CE3CB1E70B0815051710076948B892473C4D9226737C62",
        "merkleComponentHash": "1C01452EB85B4BE3F1CE3CB1E70B0815051710076948B892473C4D9226737C62",
        "index": 0
      },
      "transaction": {
        "size": 177,
        "signature": "0DE527E629B46A36B69B86117FAD287CB1903E967B23330CD5739BB176C5318578671E9BA3E7FE747D621EBE5028A71063D0D5F4DCBC96B3A532EBA96AA44C07",
        "signerPublicKey": "EF7AD177FAB80BA2D481CF16D59974A40C800F8B28960B6A14C3188930AAE3EB",
        "version": 1,
        "network": 152,
        "type": 16724,
        "maxFee": "17700",
        "deadline": "24473627704",
        "recipientAddress": "98EFE7962DE3D2A226E60F58A36D153DF3A7EFF82D21130A",
        "message": {
          "type": 0,
          "payload": ""
        },
        "mosaics": [
          {
            "id": "243661FECB8BE288",
            "amount": "100000000"
          }
        ]
      },
      "id": "5F3DF87C75AE7B73B969FF69"
    },
    {
      "meta": {
        "height": "0",
        "hash": "726151BDD6EF8B298D59C7AA5A037E12748880A5D371D3056840E491F5110213",
        "merkleComponentHash": "726151BDD6EF8B298D59C7AA5A037E12748880A5D371D3056840E491F5110213",
        "index": 0
      },
      "transaction": {
        "size": 177,
        "signature": "E2C44C4C97D1A6EA33CD01194D351F1406B000AF24D5098EA12A274D6178AFC496F4F51834493EE38230F4DB074D250C0BA6089CBC672A5FB6CCC5AE6576E40E",
        "signerPublicKey": "148E5BB506C84E6E2B96DF6371B3911566F8AA040104D751F8B15C06E30CB04C",
        "version": 1,
        "network": 152,
        "type": 16724,
        "maxFee": "600000",
        "deadline": "24473627697",
        "recipientAddress": "9815D90D4F30FC2FEAD12A67D9DAB88BB85A8AEAA5D2E589",
        "message": {
          "type": 0,
          "payload": ""
        },
        "mosaics": [
          {
            "id": "243661FECB8BE288",
            "amount": "5000000"
          }
        ]
      },
      "id": "5F3DF87C75AE7B73B969FF6A"
    }
  ],
  "pagination": {
    "totalEntries": 4,
    "pageNumber": 1,
    "pageSize": 20,
    "totalPages": 1
  }
}

**Comment by Jaguar0625** [comment]
Fixed in d516257.
